src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -582,5 +655,227 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
+
+            int indexInExistingList = 0;
+
+            bool seenNameEquals = false;
+
+            for (int i = 0; i < updatedParameters.Count; i++)
+            {
+                // Skip this parameter in list of arguments for extension method calls but not for reduced ones.
+                if (updatedParameters[i] != signaturePermutation.UpdatedConfiguration.ThisParameter
+                    || !isReducedExtensionMethod)
+                {
+                    if (updatedParameters[i] is AddedParameter addedParameter)
+                    {
+                        fullList.Add(
+                            Generator.Argument(
+                                name: seenNameEquals ? addedParameter.Name : default,
+                                refKind: RefKind.None,
+                                expression: Generator.ParseExpression(addedParameter.CallSiteValue)));
+                        separators.Add(Generator.CommaTokenWithElasticSpace());
+                    }
+                    else
+                    {
+                        if (indexInExistingList < newArguments.Count)
+                        {
+                            if (Generator.IsNamedArgument(newArguments[indexInExistingList]))
+                            {
+                                seenNameEquals = true;
+                            }
+
+                            if (indexInExistingList < newArguments.SeparatorCount)
+                            {
+                                separators.Add(newArguments.GetSeparator(indexInExistingList));
+                            }
+
+                            fullList.Add(newArguments[indexInExistingList++]);
+                        }
+                    }
+                }
+            }
+
+            // Add the rest of existing parameters, e.g. from the params argument.
+            while (indexInExistingList < newArguments.Count)
+            {
+                if (indexInExistingList < newArguments.SeparatorCount)
+                {
+                    separators.Add(newArguments.GetSeparator(indexInExistingList));
+                }
+
+                fullList.Add(newArguments[indexInExistingList++]);
+            }
+
+            if (fullList.Count == separators.Count && separators.Count != 0)
+            {
+                separators.Remove(separators.Last());
+            }
+
+            return Generator.SeparatedList(fullList, separators);
+        }
+
+        protected List<SyntaxTrivia> GetPermutedTrivia(Document document, SyntaxNode node, List<SyntaxNode> permutedParamNodes)
+        {
+            var updatedLeadingTrivia = new List<SyntaxTrivia>();
+            var index = 0;
+            SyntaxTrivia lastWhiteSpaceTrivia = default;
+
+            var lastDocumentationCommentTriviaSyntax = node.GetLeadingTrivia()
+                .LastOrDefault(t => t.HasStructure && Generator.IsDocumentationCommentTriviaSyntax(t.GetStructure()));
```

Comment:
What's the significance of the "LastOrDefault" here?

Reply:
Cool story, this local is never used. :) Gone in the next review.